test/unit/utils.js

```diff
@@ -0,0 +1,91 @@
+const { BigNumber, ethers } = require('ethers');
+const { MaxUint256, WeiPerEther } = ethers.constants;
+
+const { calculateFixedPricePremiumPerYear, calculatePremiumPerYear } = require('../../src/lib/premium-computations');
+
+const MIN_UNIT_SIZE = WeiPerEther;
+const UNIT_DIVISOR = 1000;
+const getCombinations = (size, a) => {
+  if (size === 1) {
+    return a.map(i => [i]);
+  }
+  const combinations = [];
+
+  for (let i = 0; i < a.length; i++) {
+    const smallerCombinations = getCombinations(size - 1, a.slice(i + 1));
+
+    for (const smallCombination of smallerCombinations) {
+      combinations.push([a[i], ...smallCombination]);
+    }
+  }
+  return combinations;
+};
+const getAmountSplits = (splitCount, amountInUnits) => {
+  if (splitCount === 1) {
+    return [[amountInUnits]];
+  }
+
+  const splits = [];
+  for (let i = 0; i <= amountInUnits; i++) {
+    const remainderAmount = amountInUnits - i;
+    const restOfSplits = getAmountSplits(splitCount - 1, remainderAmount);
+    for (const split of restOfSplits) {
+      splits.push([i, ...split]);
+    }
+  }
+  return splits;
+};
+
+const calculateCost = (combination, amountSplit, UNIT_SIZE, useFixedPrice) => {
+  let totalPremium = BigNumber.from(0);
+  for (let i = 0; i < combination.length; i++) {
+    const pool = combination[i];
+
+    const amount = amountSplit[i];
+
+    const amountInWei = BigNumber.from(amount).mul(UNIT_SIZE);
+
+    const premium = useFixedPrice
+      ? calculateFixedPricePremiumPerYear(amountInWei, pool.basePrice)
+      : calculatePremiumPerYear(amountInWei, pool.basePrice, pool.initialCapacityUsed, pool.totalCapacity);
+
+    totalPremium = totalPremium.add(premium);
+  }
+
+  return totalPremium;
+};
+const calculateOptimalPoolAllocationBruteForce = (coverAmount, pools, useFixedPrice) => {
```

Comment:
questionable if this should be in the code

I found it useful to check the result of the other function in complex questions without computing things by hand.